content/blog/syntax-highlighting-with-tina.md

```diff
@@ -0,0 +1,192 @@
+Once our Next.js application is running, you can navigate to [http://localhost:3000/post/HelloWorld](http://localhost:3000/post/HelloWorld). From there you will see a blog post that has a code block that looks a lot like regular text. When this is within a paragraph it may be hard for you or your editors to tell the difference between the code and regular text. Adding syntax highlighting will easily fix that!
+
+![Image without syntax highlighting](https://res.cloudinary.com/forestry-demo/image/upload/v1642686958/blog-media/syntax-highlighting-with-tina/before-image.webp)
+
+## Adding syntax highlighting
+
+### Installing our package
+
+For this tutorial we are going to use `react-syntax-highlighter` if you prefer a different package, once you make your way through this tutorial and understand how it all works, you can use your favorite package, it will work the same.
+
+```bash
+# Install the syntax highlighting package
+yarn add react-syntax-highlighter
+```
+
+Now we have our syntax highlighter installed we can create a component that we can pass our `props` to 
+
+### Creating our component
+
+First, we need to create a new filed called `Codeblock.js` inside the `components` folder. 
+
+```bash
+# create the file in the components directory
+touch  ./components/Codeblock.js
+```
+
+[Need to compare your code click here for the commit](https://github.com/tinacms/examples/commit/cdfbe238aa5510ec699f4b5066daf3dfc5fcb5ae)
+
+Inside of our newly created `Codeblock.js` file, we need to import a theme and Prism. I am going to use my favorite theme `atomOneDark` but your favorites will work just as well as mine
+
+```jsx
+import {Prism as SyntaxHighlighter} from "react-syntax-highlighter";
+import atomOneDark from "react-syntax-highlighter/dist/cjs/styles/prism/material-dark";
+```
+
+The component we are creating is going to take on two props: `children` and `language`. The children props holds all of the code we type into a code block, the language will be the language selected when editing the content for example javascript .
+
+```jsx
+const Codeblock = ({children, language}) => {
+}
+```
+
+ Now we have our component skeleton we can return our `SyntaxHighlighter` and pass in the `children` `language` .The `SyntaxHighlighter` is going to take on three different props `code`, `language` and `style` for example:
+
+```jsx
+<SyntaxHighlighter code={YOUR_CODE} language={YOUR_LANGUAGE} style={YOUR_THEME}/>
+```
+
+So for our component we are going to pass the `children` to the `code` prop and `language` to the `language` prop and the `atomOneDark` to the `style` prop. It should now look like this:
+
+```jsx
+const Codeblock = ({children, language}) => {
+    return(
+        <SyntaxHighlighter code={children || ""} language={language ||"jsx"} style={atomOneDark}/>
+    )
+}
+```
+
+You may notice that I included an, or statement to the props we are passing in. This will make sure that when we create a new empty code block that Prism doesn’t error out
```

Comment:
```suggestion
You may notice that I included an _or_ statement to the props we are passing in. This will make sure that when we create a new empty code block that Prism doesn’t error out
```